torchgen/gen.py

```diff
@@ -1478,6 +1533,7 @@ def gen_aggregated_headers(
                                 cpp_namespace="at::native",
                                 class_method_name=None,
                                 skip_dispatcher_op_registration=False,
+                                composite_graph={},
```

Comment:
Feeding empty dictionary composite graphs here is a code smell to me. (I understand why it happens; dispatch less composites can only get generated from one specific site.

Reply:
Here is what I think should be done.

You've decided to wedge the instantiation of these composites into RegisterDispatchKey. But in fact, you don't change any of the substantive logic in RegisterDispatchKey; RegisterDispatchKey is expecting a kernel definition of some form, and you're providing it. So you should give this its own `dest` module. In fact, we already have precedent for doing this: torchgen/dest/ufunc.py is in the same style, where register dispatch key expects some structured kernels to be written, and then ufunc generates the logic. This makes the factoring more modular, and fixes your passing composite graph everywhere, and it also fixes the problem where you have to suppress the predeclarations (they no longer need to be suppressed, because they're going to manage the linkage when you put it in separate files.) WDYT?

Reply:
> So you should give this its own `dest` module

Does that mean: generating a new header file with the struct declarations? Like `native_functions.py`?
If so, I think that makes sense... And it cleaner, too!

Reply:
I don't think you even need a header file! Can go straight in a cpp.